src/new_storage/backend/backend_impl.rs

```diff
@@ -0,0 +1,103 @@
+//! The general backend implementation with `openDAL`
+
+use async_trait::async_trait;
+use opendal::raw::oio::ReadExt;
+use opendal::services::{Fs, Memory};
+use opendal::{ErrorKind, Operator};
+use tokio::io::AsyncWriteExt;
+
+use super::{Backend, StorageResult};
+
+/// The `BackendImpl` struct represents a backend storage system that implements
+/// the `Backend` trait.
+#[derive(Debug)]
+pub struct BackendImpl {
+    /// The inner operator
+    operator: Operator,
+}
+
+impl BackendImpl {
+    /// Creates a new `BackendImpl` instance with the given operator.
+    pub fn new(operator: Operator) -> Self {
+        Self { operator }
+    }
+}
+
+#[async_trait]
+impl Backend for BackendImpl {
+    #[inline]
+    async fn read(&self, path: &str, buf: &mut [u8]) -> StorageResult<usize> {
+        let len = buf.len();
+        let mut reader = self.operator.reader(path).await?;
+        let mut read_size = 0;
+        loop {
+            let buf = buf
+                .get_mut(read_size..len)
+                .unwrap_or_else(|| unreachable!("The `buf` is ensured to be long enough."));
+            let result = reader.read(buf).await; // FIXME: the pointer of buffer should advance
+            match result {
+                Ok(size) => {
+                    if size == 0 {
+                        break;
+                    }
+                    read_size += size;
+                }
+                Err(e) => {
+                    // If not found just return 0.
+                    if e.kind() == ErrorKind::NotFound {
+                        break;
+                    }
+                }
+            }
+        }
+        Ok(read_size)
+    }
+
+    #[inline]
+    async fn store(&self, path: &str, buf: &[u8]) -> StorageResult<()> {
+        let mut writer = self.operator.writer(path).await?;
+        writer.write_all(buf).await?;
+        writer.close().await?;
+        Ok(())
+    }
+
+    #[inline]
+    async fn remove(&self, path: &str) -> StorageResult<()> {
+        self.operator.remove_all(path).await?;
```

Comment:
I think we should change the declaration of `remove` because it actually calls `remove_all`.

Reply:
Yes, and I can add `remove_file` to perform the removing of a whole file. 

By the way, `MockIO` trait (renamed to `Storage` in my branch, and previous `Storage` is renamed to `StorageManager` for clearer naming) does not provide `remove` method for remove a file from the storage.